app/registries/foreman/plugin.rb

```diff
@@ -372,6 +372,11 @@ def register_facet(klass, name, &block)
       @facets << Facets.register(klass, name, &block)
     end
 
+    def register_info_provider(klass)
+      @info_providers ||= []
+      @info_providers << HostInfo.register_info_provider(klass)
```

Comment:
I'd recommend avoiding the cache in `HostInfo` then. I think providers should be only cached in plugin instance and `HostInfo` should always do local + plugins. Summing few arrays should be fast enough.

the method should look like
```ruby
def register_info_provider(klass)
  @info_providers ||= []
  @info_providers << klass
  klass
end
```
and the HostInfo part:

```ruby
module HostInfo
  class << self
    def providers
      (@providers || []) + entries_from_plugins
    end

    def register_info_provider(provider)
      @providers ||= []
      @providers << provider
      provider
    end
      ...
  end
end
```